xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-image/plugin.js

```diff
@@ -326,19 +326,17 @@
                       // Clear the cache so the new image can appear on next usage of the image quick action
                       attachmentService.clearCache();
 
+                      const resourceReference = {...resource.convertEntityReferenceToResourceReference(
+                        XWiki.Model.resolve(attachment.id, XWiki.EntityType.ATTACHMENT)),
```

Comment:
I'm wondering if we shouldn't use here the ``editor.config.sourceDocument.documentReference`` as base reference, instead of the default current document, because the edited (source) document is not always the current document.